console-csharp-connect-sample/Program.cs

```diff
@@ -6,95 +6,91 @@
-                            string userInputAddress = Console.ReadLine();
-                            string messageAddress = String.IsNullOrEmpty(userInputAddress) ? mailAddress : userInputAddress;
-
-                            MailHelper.ComposeAndSendMailAsync("Welcome to Microsoft Graph development with C# and the Microsoft Graph Connect sample", Constants.EmailContent, messageAddress);
-
-                            Console.WriteLine("\nEmail sent! \n Want to send another message? Type 'y' for yes and any other key to exit.");
-                            ConsoleKeyInfo userInputSendMail = Console.ReadKey();
-                            sendMail = (userInputSendMail.KeyChar == 'y') ? true : false;
-                            Console.WriteLine();
-                        }
-                    }
-                    else
-                    {
-                        Console.ForegroundColor = ConsoleColor.Red;
-                        Console.WriteLine("We weren't able to create a GraphServiceClient for you. Please check the output for errors.");
-                        Console.ResetColor();
-                        Console.ReadKey();
-                        return;
-                    }
-
-
-
-                }
-                else
-                {
-                    Console.ForegroundColor = ConsoleColor.Red;
-                    Console.WriteLine("You haven't configured a value for ClientId in Constants.cs. Please follow the Readme instructions for configuring this application.");
-                    Console.ResetColor();
-                    Console.ReadKey();
-                    return;
-                }
-
-
-            }
-            catch (Exception ex)
-            {
-                Console.ForegroundColor = ConsoleColor.Red;
-                Console.WriteLine("Sending an email failed with the following message: {0}", ex.Message);
-                if (ex.InnerException != null)
-                {
-                    Console.WriteLine("Error detail: {0}", ex.InnerException.Message);
-                }
-                Console.ResetColor();
-                Console.ReadKey();
-                return;
-            }
-
-
-
-        }
-
-
-
-
-    }
-
-
+		private static GraphServiceClient graphServiceClient;
+		
+
+		static void Main(string[] args)
+        {	
+
+			Console.WriteLine("Welcome to the C# Console Connect Sample!\n");			
+
+			try
+			{
+				//*********************************************************************
+				// setup Microsoft Graph Client for user.
+				//*********************************************************************
+				if (!string.IsNullOrWhiteSpace(Constants.ClientId) &&
+					!string.IsNullOrWhiteSpace(Constants.Authority) &&
+					!string.IsNullOrWhiteSpace(Constants.Scopes.ToString()))
+				{
+					var graphClient = new GraphClient(Constants.ClientId, Constants.Authority, Constants.Scopes);
```

Comment:
After you have made the changes suggested above.  Change this to 

    var GraphServiceClient = GraphClientFactory.GetGraphServiceClient(Constants.ClientId, Constants.Authority, Constants.Scopes);

Reply:
Noted. Will also change the other two API calls in the MailHelper.cs class to use Task and await, if that's alright.